src/core_atmosphere/physics/physics_wrf/module_cu_ntiedtke.F

```diff
@@ -2310,17 +2612,20 @@ subroutine cuascn &
      &     pmfuq,    pmful,    pdmfup)
       is = 0
       jlm = 0
+!$acc parallel vector_length(128)
+!$acc loop vector private(zmfmax,zfac) reduction(+:is)
       do jl = 1,klon
         loflag(jl) = .false.
         zprecip(jl) = 0.
         llo1(jl) = .false.
-        is = is + klab(jl,jk+1)
+!!!     is = is + klab(jl,jk+1)
+        is = klab(jl,jk+1)
         if ( klab(jl,jk+1) == 0 ) klab(jl,jk) = 0
         if ( (ldcum(jl) .and. klab(jl,jk+1) == 2) .or.   &
              (ktype(jl) == 3 .and. klab(jl,jk+1) == 1) ) then
           loflag(jl) = .true.
-          jlm = jlm + 1
-          jlx(jlm) = jl
+!!!       jlm = jlm + 1
+!!!       jlx(jlm) = jl
```

Comment:
Can you explain why it is that these two lines can be safely commented-out?

Reply:
@Pranay-Reddy-Kommera  I can tell there's something wrong with the way I did this, but also I can't remember what the change was about. Can you look at this with me?

Reply:
Ok -- Pranay put me back on track with this. We don't want to collect the compacted index-array here
```
   2319         if ( (ldcum(jl) .and. klab(jl,jk+1) == 2) .or.   &
   2320              (ktype(jl) == 3 .and. klab(jl,jk+1) == 1) ) then
   2321           loflag(jl) = .true.
   2322           jlm = jlm + 1
   2323           jlx(jlm) = jl
   2324         end if
```
since it uses a serial loop to do it.
Instead we'll parallelize on the original Boolean array
```
   2735 ! compute the upfraft speed in cloud layer
   2736 !$acc parallel vector_length(128)
   2737 !$acc loop vector private(jl)
   2738 !!!     do jll = 1 , jlm
   2739 !!!       jl = jlx(jll)
   2740         do jl = 1, klon
   2741         if(loflag(jl)) then
```
and operate if the value had been true.
This adds some extra overhead in the non-GPU case.

I think it would be possible for us to use a prefix-sum operation to generate the index array instead but its a little messy.
I haven't worked out the details of doing this in OpenACC.
Also both the serial approach (in the CPU-only case) and the prefix-sum approach (in the parallel case) probably don't save that much time since they involve scanning the original ```loflag``` array either way.

Reply:
I just pushed up this change, it's the third commit.

Reply:
Here goes...